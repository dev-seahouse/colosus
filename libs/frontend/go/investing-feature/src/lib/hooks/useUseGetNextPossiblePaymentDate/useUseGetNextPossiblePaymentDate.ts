import type {
  InvestorBrokerageGetNextPossiblePaymentRequestDto,
  InvestorBrokerageGetNextPossiblePaymentResponseDto,
} from '@bambu/api-client';
import { TransactInvestorAuthenticatedBrokerageApi } from '@bambu/api-client';
import { useQuery } from '@tanstack/react-query';
import type { QueryArgs } from '@bambu/go-core';

const getNextPossiblePaymentDateQuery = (
  args: InvestorBrokerageGetNextPossiblePaymentRequestDto
) => ({
  queryKey: ['getNextPossiblePaymentDate', args],
  queryFn: async () => fetchNextPossiblePaymentDate(args),
});

export function useUseGetNextPossiblePaymentDate(
  args: InvestorBrokerageGetNextPossiblePaymentRequestDto,
  queryOptions?: QueryArgs<InvestorBrokerageGetNextPossiblePaymentResponseDto>
) {
  return useQuery({
    ...getNextPossiblePaymentDateQuery(args),
    ...queryOptions,
  });
}

async function fetchNextPossiblePaymentDate(
  args: InvestorBrokerageGetNextPossiblePaymentRequestDto
) {
  const brokerageApi = new TransactInvestorAuthenticatedBrokerageApi();
  const res = await brokerageApi.getNextPossiblePaymentDate(args);
  return res.data;
}

export default useUseGetNextPossiblePaymentDate;
