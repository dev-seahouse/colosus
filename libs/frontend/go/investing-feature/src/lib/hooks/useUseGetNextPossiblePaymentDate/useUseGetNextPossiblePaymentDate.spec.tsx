import useUseGetNextPossiblePaymentDate from './useUseGetNextPossiblePaymentDate';
import {
  queryClientWrapper,
  renderHook,
  waitFor,
} from '@bambu/react-test-utils';

describe('useUseGetNextPossiblePaymentDate', () => {
  it('should render successfully', async () => {
    const { result } = renderHook(
      () => useUseGetNextPossiblePaymentDate({ mandateId: 'dfdfd' }),
      {
        wrapper: queryClientWrapper,
      }
    );
    await waitFor(() => expect(result.current.isSuccess).toBe(true));
    expect(result.current.data).toMatchSnapshot();
  });
});
