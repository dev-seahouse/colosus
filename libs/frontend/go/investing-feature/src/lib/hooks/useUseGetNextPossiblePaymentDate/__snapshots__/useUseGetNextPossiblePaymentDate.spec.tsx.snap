// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useUseGetNextPossiblePaymentDate > should render successfully 1`] = `
{
  "date": "2023-08-24",
}
`;
